imvertorlite.py

```diff
@@ -1,137 +1,189 @@
+
+    tags = attr.select_one("tags").select("tag")
+    if tags:
+        attr_dict["tags"] = [{"name": tag.get("name"),
+                              "value": tag.get("value"),
+                              "notes": tag.get("notes")} for tag in tags]
+
+    return attr_dict
+
+
+def generate_schema(soup: BeautifulSoup, base: BeautifulSoup, attrs: ResultSet):
+    """
+    Converts a specific UML class to a JSON schema.
+
+    @param soup: The base soup of the XML file to resolve references.
+    @param base: The base UML class (BeautifulSoup object).
+    @param attrs: A ResultSet (list of elements) containing all the individual properties.
+    """
+
+    # Create a base schema.
+    schema = {
+        "$schema": "http://json-schema.org/draft-07/schema",
+        "$id": "https://objecttypes.vng.cloud/schema.json",
+        "title": base.get("name"),
+        # Properties are added below.
+        "properties": {},
+        # By default, don't allow for additional properties.
+        "additionalProperties": False,
+        "type": "object",
+    }
+    schema.update(
+        type_convert_dictionary(base.select_one("properties").attrs)
+    )
+    schema.update(
+        type_convert_dictionary(base.select_one("project").attrs)
+    )
+
+    schema['description'] = schema.pop('documentation')
+
+    for attr in attrs:
+        attr_dict = extract_properties(attr)
+
+        attr_name = attr_dict.get("title")
+        schema["properties"][attr_name] = attr_dict
+
+        if attr_dict.get("stereotype") == "enum":
+            enum = soup.select_one(f'element[name="enum_{attr_name}"]')
+            if enum:
+                enum_list = list(set([a.get('name') for a in enum.select("attribute")]))
+                schema["properties"][attr_name]["examples"] = [enum_list[0]]
+                schema["properties"][attr_name]["enum"] = enum_list[:]  # copy the list
+
+                # Convert the schema to JSON.
+                output = json.dumps(schema, indent=2)
+                with open(f"{schema['title'].lower()}.json", "w") as f:
+                    f.write(output)
+
+
+def process_enterprise_architect(file_name, class_name, encoding):
+    """
+    Process the enteprise architect format (XML)
+
+    @param soup: The base soup of the XML file to resolve references.
+    @param base: The base UML class (BeautifulSoup object).
+    @param attrs: A ResultSet (list of elements) containing all the individual properties.
+    """
+
+    with open(file_name, "r", encoding=encoding if encoding else "latin-1") as f:
+        data = f.read()
+
+    soup = BeautifulSoup(data, "lxml")
+
+    if class_name:
+        base_list = soup.find_all('element', {'xmi:type': 'uml:Class', 'name': class_name})
+    else:
+        # TODO: Filter out useful classes.
+        base_list = soup.find_all('element', {'xmi:type': 'uml:Class'})
+
+    for base in base_list:
+        attrs = base.select("attribute")

```

Comment:
I think we need to limit the attributes to add to the schema. See: https://github.com/open-objecten/objecttypes/pull/1/